scripts/g.extension/g.extension.py

```diff
@@ -2281,6 +2317,42 @@ def main():
         opener = urlrequest.build_opener(proxy)
         urlrequest.install_opener(opener)
 
+    # Check if authorization is supposed to be set in the request header
+    token_tags = {
+        'gitlab': "Bearer",
+        'github': "token"
+    }
+
+    # Check if url belongs to a hosting service with supported authentication
+    hosting = [token_tags[key] for key in token_tags if key in original_url]
+
+    # Get token from hardcoded environment variable
+    token = os.getenv('PERSONAL_ACCESS_TOKEN')
+
+    global HEADERS
+    # Set authorization method according to supported hosting service
+    if token and hosting:
+        HEADERS['Authorization'] = "{} {}".format(hosting[0], token)
+
+
+    # Get token from input option
+    token_input = False
+    if options['token']:
+        if options['token'] == "-":
+            token_input = sys.stdin.read().rstrip()
+        else:
+            with open(options['token'], 'r') as tokenfile:
```

Comment:
simplier `open(options['token']) as tokenfile:`

Reply:
Applied in af21fdc